stepper.c

```diff
@@ -367,16 +369,47 @@ void st_limit_check(){
   }
 
     //if limits made but not homing , servoing, or alarmed already: critical alarm.
-    if ( !(sys.state & (STATE_ALARM|STATE_HOMING)) && !(sys.state & (STATE_ALARM|STATE_FORCESERVO)) &&
+    if (!(sys.state & (STATE_ALARM | STATE_HOMING)) && !(sys.state & (STATE_ALARM | STATE_FORCESERVO)) &&
          bit_isfalse(SYS_EXEC,EXEC_ALARM)) {
       mc_reset(); // Initiate system kill.
       // Indicate hard limit critical event, print limits
       sys.alarm |= ALARM_HARD_LIMIT;
       request_report(REQUEST_LIMIT_REPORT, (EXEC_ALARM | EXEC_CRIT_EVENT));
     }
   }
+}
+
+
+// Called from ISR(TIMER4_COMPA_vect) - needs to be very efficient 
+// This checks if the desired force value is met (typically before bumping a key).
+// Once the desired force is reached, the gripper motor is stopped.
+void st_force_check()
+{
+  // Sample and filter 
+  signals_update_force();  
+
+  /* Check if desired force has been reached and
+     Stop when:
+     * Gripper is moving in positive direction
+       and the force is greater than or equal to
+       the desired force
+     OR
+     * Gripper is moving in negative direction
+       and the force is less than or equal to
+       the desired force
+  */
+  uint8_t stop = 0;
+
+  stop = ((bit_istrue(st.dir_outbits, (1 << Z_DIRECTION_BIT)) && \
+         (FORCE_VAL >= limits.bump_grip_force)) | \
+         (bit_isfalse(st.dir_outbits, (1 << Z_DIRECTION_BIT)) && \
+         (FORCE_VAL <= limits.bump_grip_force)));
```

Comment:
This needs to be broken apart and simplified.  Please assign these intermediary calculations to const values and re-write this in a more readable fashion.

Also: if a statement is wrapped in parens, you can omit the `\` at the end of the line.